src/surge-xt/gui/overlays/LuaEditors.cpp

```diff
@@ -1010,7 +1021,455 @@ struct WavetablePreviewComponent : juce::Component
+    controlArea->setBounds(0, h - controlHeight - rendererHeight - 2, w,
+                           controlHeight + rendererHeight + 2);
+
+    currentFrame->setBounds(2, h - rendererHeight - 2, 32, rendererHeight);
+    rendererComponent->setBounds(2 + 30, h - rendererHeight, w - 2 - 30, rendererHeight);
+
+    rerenderFromUIState();
+}
+
+void WavetableScriptEditor::showModulatorCode()
+{
+    preludeDisplay->setVisible(false);
+    mainEditor->setVisible(true);
+    getEditState().codeOrPrelude = 0;
+}
+
+void WavetableScriptEditor::showPreludeCode()
+{
+    preludeDisplay->setVisible(true);
+    mainEditor->setVisible(false);
+    getEditState().codeOrPrelude = 1;
+}
+
+void WavetableScriptEditor::escapeKeyPressed()
+{
+    auto c = getParentComponent();
+    while (c)
+    {
+        if (auto olw = dynamic_cast<OverlayWrapper *>(c))
+        {
+            olw->onClose();
+            return;
+        }
+        c = c->getParentComponent();
+    }
+}
+
+void WavetableScriptEditor::rerenderFromUIState()
+{
+    auto resi = controlArea->resolutionB->getSelectedId();
+    auto nfr = std::atoi(controlArea->framesT->getText().toRawUTF8());
+    auto cfr = (int)round((nfr - 1) * currentFrame->getValue()); // map slider to 0 .. nFrames - 1
+
+    auto respt = 32;
+    for (int i = 1; i < resi; ++i)
+        respt *= 2;
+
+    rendererComponent->points = Surge::WavetableScript::evaluateScriptAtFrame(
+        storage, mainDocument->getAllContent().toStdString(), respt, cfr, nfr);
+    rendererComponent->frameNumber = cfr;
+    rendererComponent->repaint();
+}
+
+void WavetableScriptEditor::generateWavetable()
+{
+    auto resi = controlArea->resolutionB->getSelectedId();
+    auto nfr = std::atoi(controlArea->framesT->getText().toRawUTF8());
+    auto respt = 32;
+    for (int i = 1; i < resi; ++i)
+        respt *= 2;
+    std::cout << "Generating wavetable with " << respt << " samples and " << nfr << " frames"
+              << std::endl;
+
+    wt_header wh;
+    float *wd = nullptr;
+    Surge::WavetableScript::constructWavetable(storage, mainDocument->getAllContent().toStdString(),
+                                               respt, nfr, wh, &wd);
+    storage->waveTableDataMutex.lock();
+    osc->wt.BuildWT(wd, wh, wh.flags & wtf_is_sample);
+    osc->wavetable_display_name = "Scripted Wavetable";
+    storage->waveTableDataMutex.unlock();
+
+    delete[] wd;
+    editor->repaintFrame();
+}
+
+void WavetableScriptEditor::sliderValueChanged(juce::Slider *slider) { rerenderFromUIState(); }
+
+std::optional<std::pair<std::string, std::string>>
+WavetableScriptEditor::getPreCloseChickenBoxMessage()
+{
+    if (controlArea->applyS->isEnabled())
+    {
+        return std::make_pair("Close Wavetable Script Editor",
+                              "Do you really want to close the wavetable editor? Any "
+                              "changes that were not applied will be lost!");
+    }
+    return std::nullopt;
+}
+
+/*
```

Comment:
No reason to leave this commented out code around is there? We have git history if we need it.